### PR TITLE
Fix:failoverbug and retries change to string type

### PR DESCRIPTION
**Pull request by hxmhlt**
**What this PR does**:
Fix:failoverbug and retries change to string type
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by pantianying** [comment]
LGTM

**Comment by fangyincheng** [comment]
Modify config.yml of `config/testdata/` pls.

**Comment by AlexStocks** [comment]
LGTM

File: cluster/cluster_impl/failover_cluster_invoker.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ package cluster_impl
 
 import (
 	perrors "github.com/pkg/errors"
+	"strconv"
```
Comment (AlexStocks) [inline cluster/cluster_impl/failover_cluster_invoker.go]: another import pls.

File: cluster/cluster_impl/failover_cluster_invoker.go [hunk 2]
```diff
@@ -53,16 +54,20 @@ func (invoker *failoverClusterInvoker) Invoke(invocation protocol.Invocation) pr
...
+	}
+	retries, err := strconv.Atoi(retriesConfig)
+	if err != nil || retries < 0 {
+		retries = constant.DEFAULT_RETRIES_INT
 	}
 	invoked := []protocol.Invoker{}
 	providers := []string{}
 	var result protocol.Result
-	for i := int64(0); i < retries; i++ {
+	for i := 0; i <= retries; i++ {
```
Comment (AlexStocks) [inline cluster/cluster_impl/failover_cluster_invoker.go]: i can not understand why using "<=" here.

**Comment by hxmhlt** [comment]
> Modify config.yml of `config/testdata/` pls.

done

File: cluster/cluster_impl/failover_cluster_invoker.go (hunk 2 above)
Comment (hxmhlt) [reply cluster/cluster_impl/failover_cluster_invoker.go]: 0 means no retries, so loop only run once.
1 means if error then run once more, so loop will run twice.

Comment (AlexStocks) [reply cluster/cluster_impl/failover_cluster_invoker.go]: got it

File: cluster/cluster_impl/failback_cluster_invoker.go [hunk 3]
```diff
@@ -54,15 +55,17 @@ func newFailbackClusterInvoker(directory cluster.Directory) protocol.Invoker {
 	invoker := &failbackClusterInvoker{
 		baseClusterInvoker: newBaseClusterInvoker(directory),
 	}
-	retriesConfig := invoker.GetUrl().GetParamInt(constant.RETRIES_KEY, constant.DEFAULT_FAILBACK_TIMES)
-	if retriesConfig <= 0 {
-		retriesConfig = constant.DEFAULT_FAILBACK_TIMES
+	retriesConfig := invoker.GetUrl().GetParam(constant.RETRIES_KEY, constant.DEFAULT_FAILBACK_TIMES)
+	retries, err := strconv.Atoi(retriesConfig)
+	if err != nil || retries < 0 {
+		retries = constant.DEFAULT_FAILBACK_TIMES_INT
```
Comment (AlexStocks) [inline cluster/cluster_impl/failback_cluster_invoker.go]: in this if cond, i think u should add a warn log here.

**Pull request merged**